src/repository.c

```diff
@@ -2691,6 +2692,11 @@ int git_repository_hashfile(
 	if (error < 0)
 		return error;
 
+	if (!p_lstat(full_path.ptr, &st) && S_ISLNK(st.st_mode)) {
```

Comment:
We're already doing a `stat` to get the file size (below).  Can we not just have a single stat for both purposes?